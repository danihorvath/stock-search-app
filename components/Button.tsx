import React from "react";

interface ButtonProps {
  onClick?: () => void;
  children: React.ReactNode;
  type?: "button" | "submit" | "reset";
}

const Button: React.FC<ButtonProps> = ({ onClick, children, type }) => {
  return (
    <button
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-md"
      onClick={onClick}
      type={type}
    >
      {children}
    </button>
  );
};

export default Button;
