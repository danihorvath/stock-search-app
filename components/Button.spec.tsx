import { expect, test, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import Button from "./Button";
import "@testing-library/jest-dom/vitest";
import * as matchers from "@testing-library/jest-dom/matchers";

expect.extend(matchers);

test("Snapshot Test", () => {
  const { container } = render(
    <Button onClick={() => {}} type="submit">
      Submit
    </Button>
  );
  expect(container).toMatchSnapshot();
});

test("Button renders correctly", () => {
  render(<Button>Click me</Button>);
  const buttonElement = screen.getByRole("button", { name: "Click me" });
  expect(buttonElement).toBeInTheDocument();
});

test("Button calls onClick handler when clicked", () => {
  const onClickMock = vi.fn();
  render(<Button onClick={onClickMock}>Click me</Button>);
  const buttonElement = screen.getByRole("button", { name: "Click me" });
  buttonElement.click();
  expect(onClickMock).toHaveBeenCalled();
});

test("Button has correct type", () => {
  render(<Button type="submit">Submit</Button>);
  const buttonElement = screen.getByRole("button", { name: "Submit" });
  expect(buttonElement).toHaveAttribute("type", "submit");
});
