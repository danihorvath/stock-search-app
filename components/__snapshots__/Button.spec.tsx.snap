// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot Test 1`] = `
<div>
  <button
    class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-md"
    type="submit"
  >
    Submit
  </button>
</div>
`;
